core/common/test/samples/InstantSamples.kt

```diff
@@ -0,0 +1,360 @@
+        check(instant1.epochSeconds == 999_999L)
+        val instant2 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = 100_123_456_789)
+        check(instant2.epochSeconds == 1_000_000 + 100L)
+        val instant3 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = -100_876_543_211)
+        check(instant3.epochSeconds == 1_000_000 - 101L)
+    }
+
+    @Test
+    fun nanosecondsOfSecond() {
+        // Getting the number of nanoseconds that passed since the start of the second
+        val instant1 = Instant.fromEpochSeconds(999_999, nanosecondAdjustment = 123_456_789)
+        check(instant1.nanosecondsOfSecond == 123_456_789)
+        val instant2 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = 100_123_456_789)
+        check(instant2.nanosecondsOfSecond == 123_456_789)
+        val instant3 = Instant.fromEpochSeconds(1_000_000, nanosecondAdjustment = -100_876_543_211)
+        check(instant3.nanosecondsOfSecond == 123_456_789)
+    }
+
+    @Test
+    fun toEpochMilliseconds() {
+        // Converting an Instant to the number of milliseconds since the Unix epoch
+        check(Instant.fromEpochMilliseconds(0).toEpochMilliseconds() == 0L)
+        check(Instant.fromEpochMilliseconds(1_000_000_000_123).toEpochMilliseconds() == 1_000_000_000_123L)
+        check(Instant.fromEpochSeconds(1_000_000_000, nanosecondAdjustment = 123_999_999)
+            .toEpochMilliseconds() == 1_000_000_000_123L)
+    }
+
+    @Test
+    fun plusDuration() {
+        // Finding a moment that's later than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
+        val fiveHoursLater = instant + 5.hours
+        check(fiveHoursLater.toEpochMilliseconds() == 12 * 60 * 60 * 1000L)
+    }
+
+    @Test
+    fun minusDuration() {
+        // Finding a moment that's earlier than the starting point by the given amount of real time
+        val instant = Instant.fromEpochMilliseconds(epochMilliseconds = 7 * 60 * 60 * 1000)
+        val fiveHoursEarlier = instant - 5.hours
+        check(fiveHoursEarlier.toEpochMilliseconds() == 2 * 60 * 60 * 1000L)
+    }
+
+    @Test
+    fun minusInstant() {
+        // Finding the difference between two instants in terms of elapsed time
+        check(Instant.fromEpochSeconds(0) - Instant.fromEpochSeconds(epochSeconds = 7 * 60 * 60) == (-7).hours)
+    }
+
+    @Test
+    fun compareToSample() {
+        // Finding out which of two instants is earlier
+        fun randomInstant() = Instant.fromEpochMilliseconds(
+            Random.nextLong(Instant.DISTANT_PAST.toEpochMilliseconds(), Instant.DISTANT_FUTURE.toEpochMilliseconds())
+        )
+        repeat(100) {
+            val instant1 = randomInstant()
+            val instant2 = randomInstant()
+            check((instant1 < instant2) == (instant1.toEpochMilliseconds() < instant2.toEpochMilliseconds()))
```

Comment:
It looks like a definition of this operation, whereas it is more nuanced. Better to use constant instants obtained from `parse`